app/javascript/containers/AdminPreviewButton.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+const consumer = window.ActionCable.createConsumer();
+
+// Not really React way but nothing can be done while the form is not in a component.
+function getSiteSettings() {
+  // We could use just the selector if naming followed a regular rule.
+  const fields = [
+    { name: 'color', selector: '#accent-color_id' },
+    { name: 'content_width', selector: '#content_width' },
+    { name: 'content_font', selector: '#content_font' },
+    { name: 'heading_font', selector: '#heading_font' },
+    { name: 'cover_size', selector: '#cover_size' },
+    { name: 'cover_text_alignment', selector: '#cover_text_alignment' },
+    { name: 'header_separators', selector: '#header_separators' },
+    { name: 'header_background', selector: '#header_background' },
+    { name: 'header_transparency', selector: '#header_transparency' },
+    { name: 'header_login_enabled', selector: '#header_login_enabled' },
+    { name: 'footer_background', selector: '#footer_background' },
+    { name: 'footer_text_color', selector: '#footer_text_color' },
+    { name: 'footer_links_color', selector: '#footer-links-color' }
+  ];
+
+  const values = fields.reduce((acc, field) => ({
+    ...acc,
+    [field.name]: document.querySelector(field.selector).value
+  }), {});
+
+  const headerCountryColors = Array.from(document.querySelectorAll('.country-colors-container input[type="color"]'))
+    .map(countryColor => countryColor.value).join(' ');
+
+  return { ...values, 'header-country-colours': headerCountryColors };
+}
+
+export default function AdminPreviewButton({ className, text, host, slug }) {
+  useEffect(() => {
+    consumer.subscriptions.create('PreviewChannel', {
+      connected: () => {},
+      disconnected: () => {},
+      received: () => {
+        setPreviewButtonState({ ...previewButtonState, isLoading: false });
+
+        // Open preview page
+        window.open(`${host}/admin/sites/${slug}/preview`, '_blank');
+      }
+    })
+    return () => {};
+  }, []);
+  const [previewButtonState, setPreviewButtonState] = useState({ isLoading: false });
```

Comment:
I think this should be before the `useEffect` call because it uses `previewButtonState`.

Reply:
Also, `previewButtonState` doesn't have to be an object. Why not storing the boolean directly?